pennylane/ops/qubit/observables.py

```diff
@@ -106,17 +106,37 @@ def eigvals(self):
         """
         return self.eigendecomposition["eigval"]
 
+    @staticmethod
+    def compute_diagonalizing_gates(eigenvectors, wires):
+        """Diagonalizing gates of this operator.
+
+        Args:
+            eigenvectors (array): eigenvectors of this operator, as extracted from op.eigendecomposition["eigvec"]
+            wires (Iterable): wires that the operator acts on
+
+        Returns:
+            list[.Operator]: list of diagonalizing gates
+
+        **Example**
+
+        >>> A = np.array([[-6, 2 + 1j], [2 - 1j, 0]])
+        >>> _, evecs = np.linalg.eigh(A)
+        >>> qml.Hermitian.compute_diagonalizing_gates(evecs, wires=[0])
+        [QubitUnitary(tensor([[-0.94915323-0.j,  0.2815786 +0.1407893j ],
+                              [ 0.31481445-0.j,  0.84894846+0.42447423j]], requires_grad=True), wires=[0])]
```

Comment:
🙌

Reply:
Yes, this one of the few cases where the init method does some computation that we don't want to loose. At the moment I am solving this by changing what gets put into the `compute` method...I am in two minds about this though.

Reply:
yeah same. I must admit, if I was coding up the class myself, my intuition would be to do exactly what the templates do; process in init, and then store in `hyperparameters`. This is partly because `hyperparameters` is not user-facing; it _feels_ like a place where developers can store the 'real' hyperparameters lol